File: backend/src/models/Expense.js
const mongoose = require('mongoose');

// A miscellaneous cash withdrawal — money taken out of hand for something
// other than a gold purchase, sale, or loan (e.g. personal use, shop rent).
const expenseSchema = new mongoose.Schema(
  {
    date: { type: Date, required: true },
    amount: { type: Number, required: true, min: 0.01 },
    note: { type: String, default: '', trim: true },
  },
  { timestamps: true }
);

expenseSchema.index({ date: 1, createdAt: 1 });

module.exports = mongoose.model('Expense', expenseSchema);

File: mobile/lib/models/expense.dart
class Expense {
  const Expense({
    required this.id,
    required this.date,
    required this.amount,
    required this.note,
    this.createdAt,
  });

  final String id;
  final DateTime date;
  final double amount;
  final String note;
  final DateTime? createdAt;

  factory Expense.fromJson(Map<String, dynamic> j) => Expense(
        id: j['id'] as String,
        date: DateTime.parse(j['date'] as String),
        amount: (j['amount'] as num).toDouble(),
        note: (j['note'] as String?) ?? '',
        createdAt: j['createdAt'] != null
            ? DateTime.tryParse(j['createdAt'] as String)
            : null,
      );
}
